### PR TITLE
Update upload media errors for published/scheduled/etc..

### LINKED ISSUE #9933: After publishing while offline, the "retry" button on the Post List button will publish the Post
Imported from this comment https://github.com/wordpress-mobile/WordPress-Android/issues/9555#issuecomment-492858217

> This is how things currently work right now:
> 
> - Android: Publishing while offline and retrying will lead to the post being uploaded and **published**
> - iOS: Publishing while offline and retrying will lead to the post being uploaded as a draft
> 
> I'm not sure which one is right but Android's behavior is scary for me as a user. 

## Decision
As the user intended to publish the post, retrying should also publish the post
To clarify, we should amend the error state to mention posting. Error message should say "Could not upload media. Post not published." Then its clear that retry relates to posting, not just media.
To make this work we need to differentiate the error state in publishing/saving/scheduling scenarios. 

1. User publishes and media upload error (use case in this issue)
Error Message: "Could not upload media. Post not published."
Action and behavior: Retry. Retrying publishes the post if media uploads successfully

2. User schedules and media upload error (scheduling rather than publishing)
Error Message: "Could not upload media. Post not scheduled."
Action and behavior: Retry. Retrying schedules the post if media uploads successfully

3. User saves and media upload error (user did not intend to publish)
Error Message: "Could not upload media."
Action and behavior: Retry. Retrying only retries uploading the media as the user has not changed the post state

## Related Issues

https://github.com/wordpress-mobile/WordPress-iOS/issues/11420

### PR DESCRIPTION
**Pull request by maxme**
Fixes #9933  by changing the error message after media upload failure.

Test 1:
- Create a new post.
- Add a media and make the upload fail (airplane mode for example).
- Go back (draft), check the message is "Could not upload media"

Test 2:
- Create a new post.
- Add a media and make the upload fail (airplane mode for example).
- Tap publish, check the message is "Could not upload media. Post not published."

Do the same test for private, scheduled and pending posts and notice different error message.

Note: I first wanted to compose the error message using the base "Could not upload media" and concatenate the specific error, but it's often source of issues in translations and should be solved by translation memory.

Before | After
-|-
ff395b985733|Screenshot_1562597183

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1533,7 +1533,10 @@
     <string name="error_media_load">Unable to load media</string>
     <string name="error_media_save">Unable to save media</string>
     <string name="error_media_canceled">Uploading media were canceled</string>
-    <string name="error_media_recover_post">Unable to upload this post\'s media</string>
+    <string name="error_media_recover_post">Could not upload media</string>
```
Comment (shiki) [inline WordPress/src/main/res/values/strings.xml]: ⚠️ We're missing a period in the sentence. 🙂

File: WordPress/src/test/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelperTest.kt [hunk 2]
```diff
@@ -179,6 +180,36 @@ class PostListItemUiStateHelperTest {
                 post = createPostModel(isLocallyChanged = true, status = POST_STATE_PRIVATE),
                 uploadStatus = createUploadStatus(uploadError = UploadError(MediaError(AUTHORIZATION_REQUIRED)))
         )
+        assertThat(state.data.statuses).contains(UiStringRes(R.string.error_media_recover_post_not_published))
```
Comment (shiki) [inline WordPress/src/test/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelperTest.kt]: TIL there's also a `containsOnly` matcher so we can avoid the `contains` + `hasSize(1)` dance. 🎉 

```kt
        assertThat(state.data.statuses).containsOnly(UiStringRes(R.string.error_media_recover_post))
```

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (maxme) [reply WordPress/src/main/res/values/strings.xml]: I didn't use a period here following https://material.io/design/communication/writing.html#principles (see "Skip unnecessary punctuation"). I'm not sure about it, feels a bit weird when you see 2 different messages.

@osullivanchris I'm not sure if we want to follow material design strictly here (or if we have WP guidelines for writing).

File: WordPress/src/test/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelperTest.kt (hunk 2 above)
Comment (maxme) [reply WordPress/src/test/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelperTest.kt]: Thanks! updated in b27868a

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (shiki) [reply WordPress/src/main/res/values/strings.xml]: Thank you for the link. It looks like you are right and it seems consistent with the other error messages above this.

Comment (shiki) [reply WordPress/src/main/res/values/strings.xml]: @osullivanchris, we'd love to hear what you think. 🙂

**Comment by osullivanchris** [comment]
> I also have this question, probably for you and @osullivanchris. When I try to upload a scheduled post with a media, the Post List card has the correct message. However, a snackbar is also shown at the same time with a different error message. Is this something we should be concerned about?

Good spot. In #9555 we're changing the behaviour for offline publishing/scheduling. In that update we'll be changing the snackbar message. Also discussed in paCBwp-bF-p2. Right now this snack bar is for multiple cases. With #9555 we'll have different snackbars for different cases. So lets work on it then and we can make this one specific to media upload error, and make it similar/same to the message shown on the post.

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (osullivanchris) [reply WordPress/src/main/res/values/strings.xml]: Lets leave it without the period as all the other errors are the same for now. 

For future...
I checked our style guide and couldn't find something specific. The closest thing I saw was 
"Use periods at the end of list items that consist of complete sentences."
Not everyone would use periods in list items I imagine. So it seems like we lean on the side of using a period if that example is anything to go by.

Separately we could update them all to have periods if we think its better, but lets leave it for this issue. I think @jklausa might have been doing a similar inventory for iOS.

**Pull request merged**